### PR TITLE
CDD-2592: tweaked error msgs. Missing mrn parameter returns "Invalid Search" in message element as does an mrn that does not pass length validation.

### PR DESCRIPTION
**Pull request by paulblakehmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/declarations/information/controllers/StatusController.scala [hunk 1]
```diff
@@ -42,8 +42,14 @@ class StatusController @Inject()(val validateAndExtractHeadersAction: ValidateAn
...
-    val searchType = Mrn(mrn)
-    implicit asr: AuthorisedRequest[AnyContent] => search(searchType)
+      val searchType = Mrn(mrn)
+      implicit asr: AuthorisedRequest[AnyContent] =>
+        if (mrn.isEmpty) {
+          logger.error("Missing mrn")
+          Future.successful(ErrorResponse(NOT_FOUND, NotFoundCode, "Invalid Search").XmlResult.withConversationId)
+        } else {
+          search(searchType)
+      }
```
Comment (glidester) [inline app/uk/gov/hmrc/customs/declarations/information/controllers/StatusController.scala]: One unnecessary level of indentation. Also, should this check also be added for the other three search types (even though they are not going live yet)?

File: test/unit/controllers/StatusControllerSpec.scala [hunk 2]
```diff
@@ -102,10 +102,7 @@ class StatusControllerSpec extends UnitSpec
 
   private val errorResultBadgeIdentifier = errorBadRequest("X-Badge-Identifier header is missing or invalid").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
   private val errorResultMissingIdentifiers = errorBadRequest("Both X-Submitter-Identifier and X-Badge-Identifier headers are missing").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
-  private val errorResultInvalidMrn = errorBadRequest("Invalid MRN").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
-  private val errorResultInvalidDucr = errorBadRequest("Invalid DUCR").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
-  private val errorResultInvalidUcr = errorBadRequest("Invalid UCR").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
-  private val errorResultInvalidInventoryReference = errorBadRequest("Invalid InventoryReference").XmlResult.withHeaders(X_CONVERSATION_ID_HEADER)
```
Comment (glidester) [inline test/unit/controllers/StatusControllerSpec.scala]: Will these not be useful when a user sends one of these search parameters (non-empty content) but the values are invalid?

Comment (paulblakehmrc) [reply test/unit/controllers/StatusControllerSpec.scala]: Programme doesn't want that text in any error msg.

File: app/uk/gov/hmrc/customs/declarations/information/controllers/StatusController.scala (hunk 1 above)
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/customs/declarations/information/controllers/StatusController.scala]: Force pushed that change. 
I'll add them later for the other types when needed. Just now I want to test mrn in a managed env.

**Pull request merged**